docs/Design/SoftDetailedDes/MIS.tex

```diff
@@ -630,6 +630,150 @@ \subsubsection{Local Functions}
 
 \newpage
 
+\section{MIS of Users Module} \label{mUsers} 
+The Users Module exists to handle CRUD operations for operations related to users and their grocery trips.
+
+\subsection{Module}
+
+users
+
+\subsection{Uses}
+
+ Database Driver Module (Section \ref{mDBDriver}) 
+
+\subsection{Syntax}
+
+\subsubsection{Exported Constants}
+\textbf{Item} as defined in Section \ref{mExtraction} \\
+
+\noindent \textbf{GroceryTrip} as defined in Section \ref{mExtraction} \\
+
+\noindent \textbf{User}
+
+\begin{table}[H]
+  \begin{tabular}{|p{0.25\textwidth}|p{0.25\textwidth}|p{0.40\textwidth}|}
+    \hline
+    \textbf{Output Name} & \textbf{Output Type} & \textbf{Description} \\
+    \hline
+    userId & Integer & Unique user id number \\
+    \hline
+    email & String & Email of user \\
+    \hline
+    password & String & Password of user (Encrypted) \\
+    \hline
+    firstName & String & First name of User \\
+    \hline
+    lastName & String & Last name of User \\
+    \hline
+    location & String & Location of user's home-base. (Coordinates) \\
+    \hline
+  \end{tabular}
+\end{table}
+
+\subsubsection{Exported Access Programs}
+
+\begin{center}
+\begin{tabular}{p{5cm} p{3cm} p{3cm} p{5cm}}
+\hline
+\textbf{Name} & \textbf{In} & \textbf{Out} & \textbf{Exceptions} \\
+\hline
+newUser & User & - & userExistsException \\
+getUser & String & User & userDneException \\
+updateUser & User & - & userDneException \\
+deleteUser & String & - & userDneException \\
+addTrip & GroceryTrip & - & userDneException \\
+getTrip & DateTime & GroceryTrip & - \\
+\hline
+\end{tabular}
+\end{center}
+
+\subsection{Semantics}
+
+\subsubsection{State Variables}
+None
+
+\subsubsection{Environment Variables}
+None 
+
+\subsubsection{Assumptions}
+None
+
+\subsubsection{Access Routine Semantics}
+
+\noindent newUser(\textit{user}):
+\begin{itemize}
+\item transition: \( \neg userExists(\textit{user.userId}) \Rightarrow addUserInDB(\textit{user})\) 
+\item output: none
+\item exception \( \textit{exc} \coloneqq userExists(\textit{user.userId}) \Rightarrow \\ userExistsException\)
+\end{itemize}
+
+\noindent getUser(\textit{userId}):
+\begin{itemize}
+\item transition: none
+\item output: \( userExists(\textit{user.userId}) \Rightarrow getUserInDB(userId)\)
+\item exception \( \textit{exc} \coloneqq \neg userExists(\textit{user.userId}) \Rightarrow \\ userDneException\)
+\end{itemize}
+
+\noindent updateUser(\textit{user}):
+\begin{itemize}
+\item transition: none
+\item output: \( userExists(\textit{user.userId}) \Rightarrow \) user, from database (Section \ref{mDBDriver}).
```

Comment:
updateUser doesn't have an output in the above table but it does here. Did you mean to put this in transition or is updateUser supposed to have an output?